ops/terraform/services/pipeline/modules/bfd_pipeline_manager/README.md

```diff
@@ -0,0 +1,73 @@
+# `bfd_pipeline_manager` Sub-module
+
+This submodule defines a Lambda and its corresponding infrastructure. This Lambda,
+`bfd-${env}-pipeline-manager`, "manages" the BFD Pipeline Terraservice; it is invoked when new files
+are created in the `env`'s corresponding S3 pipeline/ETL bucket at specific paths:
+
+- `Synthetic/Incoming/`
+- `Synthetic/Done/`
+- `Incoming/`
+- `Done/`
+
+Objects uploaded to or moved to these paths in the ETL bucket will invoke this Lambda, which will
+then determine if the BFD CCW Pipeline should be started or not. This depends on a few factors:
+
+- If the file was uploaded to either of the `Incoming` folders, and there are no _ongoing_ or
+  _queued_ data loads _at all_ (including other groups), the `bfd-deploy-pipeline-terraservice`
+  Jenkins pipeline is started using the `bfd-job-broker` with the arguments set to _create_ the CCW
+  Pipeline instance
+- Else, if the file was uploaded to either of the `Done` folders, there are no files in the
+  corresponding `Incoming` folder, all non-optional RIF files are present in the `Done` folder,
+  _and_ there are no ongoing loads _at all_ (including other groups), the
+  `bfd-deploy-pipeline-terraservice` Jenkins pipeline is started using the `bfd-job-broker` with the
+  arguments set to _destroy_ the CCW Pipeline instance
```

Comment:
Just a note that this may not work correctly with synthea data; we will have folders sitting in synthetic/incoming that will not be processed until their week comes due (and we may have up to 2 years worth of these weekly loads sitting and waiting in synthetic/incoming). Depending on the check it may never be able to destroy the instance due to these folders sitting in the incoming folder as expected.

Reply:
We discussed this offline, but essentially this can be solved by scheduling runs of `bfd-pipeline-manager` Lambda based upon the folder timestamp of future data loads. This PR does not do this, so it would be susceptible to running indefinitely if many future loads were uploaded to the `Synthetic/Incoming` folder.